lib/rspec/openapi/record_builder.rb

```diff
@@ -55,7 +55,10 @@ def safe_parse_body(response, media_type)
   def extract_headers(request, response)
     request_headers = RSpec::OpenAPI.request_headers.each_with_object([]) do |header, headers_arr|
       header_key = header.gsub('-', '_').upcase.to_sym
-      header_value = request.get_header(['HTTP', header_key].join('_')) || request.get_header(header_key)
+
+      header_value = request.get_header(['HTTP', header_key].join('_')) ||
+                     request.get_header(header_key) ||
+                     request.get_header(header_key.to_s)
```

Comment:
I found one problem with the new edits - https://github.com/exoego/rspec-openapi/pull/198/files#diff-a79ceb062f9ec26b6203f0b7541bd0a9b1cc2ebf3370cac3da9039f4f11dc1a7R56 - Rack test seems to send strings instead of characters in headers, so because of this edit, the headers are no longer defined.